CRM/MembersOnlyEvent/Form/MembersOnlyEventTab.php

```diff
@@ -331,4 +353,85 @@ private function downgradeToNormalEvent($membersOnlyEventID) {
     $membersOnlyEvent->delete();
   }
 
+  /**
+   * Gets price fields from an event ID
+   * Empty Price Fields will be returned if an event is not paid event
+   * or price set is not been used or one of the price fields
+   * is not Radio or Checkbox
+   *
+   * @return array
+   * @throws CiviCRM_API3_Exception
+   */
+  private function getEventPriceFieldValues() {
+    $priceFieldValueLabelsByIDs = [];
+    $priceSetId = CRM_Price_BAO_PriceSet::getFor('civicrm_event', $this->_id, NULL);
+
+    if (!$this->isPaidEvent() || !$priceSetId) {
+      return $priceFieldValueLabelsByIDs;
+    }
+
+    $priceFields = civicrm_api3('PriceField', 'get', [
```

Comment:
From lines 373 to lines 397, we can just use one API call to get price field values from price set ID 

Something like this 

```
`civicrm_api3('PriceFieldValue', 'get', [
  'sequential' => 1,
  'price_field_id.price_set_id' => $priceSetID,
]);`
```

Reply:
We couldn't do that because we need to check price field with `isPriceFieldValidForMembersOnlyEvent`

Reply:
@ahed-compucorp Actually we can, the example below is to get price field html type from price set ID from price field value API. You would need to know what fields you want. 

 ```
civicrm_api3('PriceFieldValue', 'get', [
  'sequential' => 1,
  'return' => ["price_field_id.html_type"],
  'price_field_id.price_set_id' => $priceSetID,,
]);
```

I think it is cleaner, and we don't need to call API twice.

Reply:
yeah